Sources/Stagehand/AnimatableProperty/AnimatableProperty+CGAffineTransform.swift

```diff
@@ -0,0 +1,231 @@
+            m12: CGFloat.value(between: initialDecomposition.m12, and: finalDecomposition.m12, at: progress),
+            m21: CGFloat.value(between: initialDecomposition.m21, and: finalDecomposition.m21, at: progress),
+            m22: CGFloat.value(between: initialDecomposition.m22, and: finalDecomposition.m22, at: progress)
+        ).recompose()
+    }
+
+}
+
+// MARK: -
+
+extension CGAffineTransform {
+
+    // This logic is based on functionality in WebKit, which has the following copyrights:
+    //   Copyright (C) 2005, 2006, 2013 Apple Inc.  All rights reserved.
+    //   Copyright (C) 2009 Torch Mobile, Inc.
+
+    // MARK: - Internal Methods
+
+    func decomposed() -> DecomposedMatrix {
+        if isIdentity {
+            return .init()
+        }
+
+        var matrix = self
+        var decomposedMatrix = DecomposedMatrix()
+
+        // Affine transforms are represented by a 3x3 matrix:
+        //
+        //   ┌─         ─┐
+        //   │ a   b   0 │
+        //   │           │
+        //   │ c   d   0 │
+        //   │           │
+        //   │ tx  ty  1 │
+        //   └─         ─┘
+        //
+        // The `a`, `b`, `c`, and `d` elements are made from a combination of applied scale, rotation, and shear
+        // transforms. The `tx` and `ty` elements represent the translation.
+
+        // Translation
+        //
+        //   ┌─         ─┐┌─           ─┐   ┌─                                       ─┐
+        //   │ 1   0   0 ││  a    b   0 │   │         a                  b          0 │
+        //   │           ││             │   │                                         │
+        //   │ 0   1   0 ││  c    d   0 │ = │         c                  d          0 │
+        //   │           ││             │   │                                         │
+        //   │ tx  ty  1 ││ tx0  ty0  1 │   │ tx·a + ty·c + tx0  tx·b + ty·d + ty0  1 │
+        //   └─         ─┘└─           ─┘   └─                                       ─┘
+
+        // Since translation only affects the `tx` and `ty` elements, we can store those directly.
+        decomposedMatrix.translateX = matrix.tx
+        decomposedMatrix.translateY = matrix.ty
+
+        // Scaling
+        //
+        //   ┌─         ─┐┌─         ─┐   ┌─             ─┐
+        //   │ sx  0   0 ││ a   b   0 │   │ sx·a  sx·b  0 │
+        //   │           ││           │   │               │
+        //   │ 0   sy  0 ││ c   d   0 │ = │ sy·c  sy·d  0 │
+        //   │           ││           │   │               │
+        //   │ 0   0   1 ││ tx  ty  1 │   │  tx    ty   1 │
+        //   └─         ─┘└─         ─┘   └─             ─┘
+
+        decomposedMatrix.scaleX = hypot(matrix.a, matrix.b)
+        decomposedMatrix.scaleY = hypot(matrix.c, matrix.d)
+
+        let determinant = matrix.a * matrix.d - matrix.b * matrix.c
+
+        // If determinant is negative, one axis was flipped.
+        if determinant <= 0 {
```

Comment:
This differs a bit from WebKit's implementation of decomposition in the way it handles a determinant of zero. From my test cases, I found that this gave me better results. May have to tweak this in the future if we come up with more cases where this doesn't work.